pennylane/transforms/insert_ops.py

```diff
@@ -210,9 +211,15 @@ def f(w, x, y, z):
     """
     # decompose templates and their adjoints (which fixes a bug in the tutorial_error_mitigation demo)
     # TODO: change this to be cleaner and more robust
-    circuit = circuit.expand(
-        stop_at=lambda op: not hasattr(qml.templates, op.name) and not isinstance(op, Adjoint)
-    )
+    try:
+        circuit = circuit.expand(
+            stop_at=lambda op: not hasattr(qml.templates, op.name) and not isinstance(op, Adjoint)
+        )
+    except qml.QuantumFunctionError as e:
+        raise qml.QuantumFunctionError(
+            "The insert transform cannot transform a circuit with non-commuting observables. "
+            "Consider wrapping the gates in their own function and transforming only that function."
```

Comment:
Added "measure" as one can still apply non-commuting unitary observables as gates in the circuit and use insert without error. 

Also what do you mean by consider wrapping the gates in their own function? 
```suggestion
            "The insert transform cannot transform a circuit measuring non-commuting observables. "
            "Consider wrapping the gates in their own function and transforming only that function."
```

Reply:
hmm, was hoping that language was clear but maybe you can help. I basically mean to do what was in the original bug report (that did it within the quantum function, but you can separate it out if you prefer the decorator notation). My drop-down example in the PR description shows what I mean : ) a user will do what's in the first drop-down, get that error, then hopefully (maybe not tbh 😬) do what's in the second drop-down.